opendevin/action/bash.py

```diff
@@ -1,17 +1,23 @@
+import os
+import pathlib
 from dataclasses import dataclass
 from typing import TYPE_CHECKING
 
 from .base import ExecutableAction
-from opendevin.schema import ActionType
+from opendevin import config
+from opendevin.schema import ActionType, ConfigType
 
 if TYPE_CHECKING:
     from opendevin.controller import AgentController
     from opendevin.observation import CmdOutputObservation
 
+from opendevin.observation import IPythonRunCellObservation
+
 
 @dataclass
 class CmdRunAction(ExecutableAction):
     command: str
+    thought: str = ''
```

Comment:
Why is there a `thought` on CmdRunAction?

Reply:
I've been thinking we should let the LLM override `message` on all the actions it produces, if that helps here

Reply:
Because for each bash, CodeActAgent will produce a corresponding thought there.. I think it will be pretty messy if we need to emit two action at the same time (ThoughtAction followed by `CmdRun`), so i just added a `thought` here accompanying CmdRun..

Reply:
well, i was thinking even if we want to use `.message` to covey `thought`, we'd still need an additional argument called `thought` or `message` so we can implement the `.message` attribute appropriately